hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigurationAwareConfig.java

```diff
@@ -1012,6 +1013,48 @@ public Config setCacheEventJournalConfigs(Map<String, EventJournalConfig> eventJ
         throw new UnsupportedOperationException("Unsupported operation");
     }
 
+    @Override
+    public Map<String, ReliableIdGeneratorConfig> getReliableIdGeneratorConfigs() {
+        Map<String, ReliableIdGeneratorConfig> staticMapConfigs = staticConfig.getReliableIdGeneratorConfigs();
+        Map<String, ReliableIdGeneratorConfig> dynamicMapConfigs = configurationService.getReliableIdGeneratorConfigs();
+        return aggregate(staticMapConfigs, dynamicMapConfigs);
+    }
+
+    @Override
+    public ReliableIdGeneratorConfig findReliableIdGeneratorConfig(String name) {
+        return getReliableIdGeneratorConfigInternal(name, "default").getAsReadOnly();
+    }
+
+    @Override
+    public ReliableIdGeneratorConfig getReliableIdGeneratorConfig(String name) {
+        return getReliableIdGeneratorConfigInternal(name, name);
+    }
+
+    private ReliableIdGeneratorConfig getReliableIdGeneratorConfigInternal(String name, String fallbackName) {
+        String baseName = getBaseName(name);
+        Map<String, ReliableIdGeneratorConfig> staticMapConfigs = staticConfig.getReliableIdGeneratorConfigs();
+        ReliableIdGeneratorConfig config = lookupByPattern(configPatternMatcher, staticMapConfigs, baseName);
+        if (config == null) {
+            config = configurationService.findReliableIdGeneratorConfig(baseName);
+        }
+        if (config == null) {
+            config = staticConfig.getReliableIdGeneratorConfig(fallbackName);
+        }
+        return config;
+    }
+
+    @Override
+    public Config addReliableIdGeneratorConfig(ReliableIdGeneratorConfig config) {
```

Comment:
This method should be also overridden on the client-side dynamic config support class `ClientDynamicClusterConfig`. You will need to add a method in `DynamicConfigTemplate` on client-protocol side for that, so since this PR is already big enough, feel free to add client-side dynamic config support in a further PR.

Reply:
ugh, the whole damn config thing was like 25% of the time I spent on the entire feature...